noir-projects/aztec-nr/aztec/src/state_vars/public_immutable.nr

```diff
@@ -14,9 +14,9 @@ pub struct PublicImmutable<T, Context> {
 }
 // docs:end:public_immutable_struct
 
-impl<T, Context, let N: u32> Storage<N> for PublicImmutable<T, Context>
+impl<T, Context, let M: u32, let N: u32> Storage<N> for PublicImmutable<T, Context>
 where
-    T: Packable<N>,
+    WithHash<T, M>: Packable<N>,
```

Comment:
Let's add a `PublciImmutable` with a struct to the test contract and test in the storage layout that it allocates N+1 values

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/8022/commits/10bcbd6effc43c4b0bdd6dbe413e89dec4242ed8